main.tf

```diff
@@ -178,7 +178,8 @@ module "eks" {
   ssh_pvt_key_path             = local.ssh_pvt_key_path
   bastion_user                 = local.bastion_user
   bastion_public_ip            = try(module.bastion[0].public_ip, "")
-  kms_key                      = local.kms_key_arn
+  cluster_kms_key              = var.use_kms ? (var.kms_key_id == null ? resource.aws_kms_key.domino[0].arn : data.aws_kms_key.key[0].arn) : aws_kms_key.eks_cluster[0].arn
```

Comment:
this can be simplified since you already computed part of this as the kms_key_arn.
you could also make it more brain dead with a `try(aws_kms_key.eks_cluster[0].arn, local.kms_key_arn)`